Add new files to the folder

Tashwin opened the pull request.

Commit:
added image support and some models in simulation

Commit:
added image support and some models in simulation

Tashwin committed:
Delete README

caguero commented:
Running the example.py I got this error:

``` python
Traceback (most recent call last):
  File "example.py", line 7, in <module>
    roadPointWidthMap = osmRoads.getRoadCoords()
AttributeError: Osm2Dict instance has no attribute 'getRoadCoords'
```

It seems that the class Osm2Dict does not have a getRoadCoords() method.

caguero commented:
I recommend to follow [pep8 style guide](http://www.python.org/dev/peps/pep-0008/). What I normally do is install a pep8 linter that automatically hightlights all the instructions that are not pep8 compliant. Here is [an example](https://github.com/SublimeLinter/SublimeLinter-pep8) for SublitmeText editor.

Commit:
image support and some simulated models

Commit:
Merge branch 'master' of https://github.com/Tashwin/gazebo_osm-1

Tashwin commented:
I just added example.py to the git account. This is updated and tested.

Tashwin commented:
Thanks I'll look it up and install pep8 linter

wjwwood commented:
Also, you can setup an automated test for pep8, like I did for bloom:

https://github.com/ros-infrastructure/bloom/blob/master/test/test_code_format.py

Tashwin commented:
Thanks! that helped a lot ! 
I'll make the files pep8 compliant and make a fresh commit.

Commit:
pep8 compliant; added buildings and amenities

Tashwin committed:
Delete outFile.sdf

Tashwin committed:
Delete mymap.png

caguero commented:
Not a big deal but if you manage to keep all the lines with a maximum of 79 chars, all your files will have zero pep8 warnings/errors.

I see a sampleMapImage.png file. That should be removed from the repo. It will be generated on demand by your script.

Tashwin committed:
Delete sampleMapImage.png

Tashwin commented:
I just deleted the sampleMapImage.png files. I will also update the other files and commit the changes to the repo.

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,281 @@
+            startCoords = raw_input("Enter starting coordinates: [lon lat] :").split(' ')
+            endCoords = raw_input("Enter ending coordnates: [lon lat]: ").split(' ')
+
+            if startCoords != [] and endCoords != [] and len(startCoords) == 2 and len(endCoords) == 2:
```
You could type:

``` python
  if startCoords
```

instead of:

``` python
 if startCoords != []
```

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,281 @@
+
+class Osm2Dict:
+
+    def __init__(self):
```
I'd suggest to add latStart, lonStart, latStop and lonStop as parameters of the constructor. You could move the interaction with the user to a main program.

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,281 @@
+        return point
+
+    def getMapDetails(self):
+        ''' Returns a list of highways with corresponding widhts and a list of all the models to be included'''
```
Typo: widhts <--> widths

caguero commented:
We could remove sampleOutput.sdf too, if is generated by a script

caguero commented:
It would be nice if every class has a test that checks its correct operation. Take a look at a [unit test module](http://docs.python.org/2/library/unittest.html) for an ever better approach.

Tashwin commented:
Yes, that was my next question. If I should make the programs, command line compatible for input arguments. 
Next, for now I simply included the image generation as part of the osm2dict program, is that fine?
Also, I will look into the unit testing modules.

Tashwin committed:
Delete sampleOutput.sdf

caguero commented:
Yes, the image will be generated in your osm2dict, that's perfect. 

Try to isolate the class that implements some functionality from the program that tests it. The program that tests your class/es might be interactive (as it is right now) or could use command line arguments. If you want to add support for command line arguments take a look at [argparse module](http://docs.python.org/dev/library/argparse.html).

Tashwin commented:
Ok. I'll incorporate all the changes. 
So, I will get rid of all the interaction happening from osm2dict, and do it from a main test program like example.py.
I have used argparse module before, so I'm familiar with its working and can incorporate it in  our test program or keep it  somewhat similar to as it is now. So, I think I'll I push this back and work on building unit tests first?

caguero commented:
Sounds good.

scpeters commented:
Regarding the image generation in osm2dict, I think it might be better to put it in a separate file, in order to separate the dependencies. If someone just wants to get osm data, they have to install mapnik if they're both in the same file.

Tashwin commented:
Ok that has a file for itself. I can do that simply enough.

Commit:
pep8 compliant; moved user interaction to example.py

Commit:
Merge branch 'master' of https://github.com/Tashwin/gazebo_osm-1

Tashwin commented:
I added a new file which goes in and downloads the actual osm data, which is then used by both, the getMapImage.py and osm2dict.py. This prevents downloading same data twice.
Should I add a clean-up function which deletes the image and .osm files created during the build?

Commit:
Osm file downloaded seperately

caguero commented on getMapImage.py:
```diff
@@ -0,0 +1,76 @@
+
+        m.append_style(highwayType, styleType)
+
+    ds = mapnik.Osm(file = osmFile)
```
Remove spaces around parameter equals

caguero commented on getOsmFile.py:
```diff
@@ -0,0 +1,24 @@
+import osmapi
+
+
+def getOsmFile(box, filename = 'map.osm'):
```
Remove spaces around parameter equals

caguero commented on getOsmFile.py:
```diff
@@ -0,0 +1,24 @@
+    osmFile = urllib2.urlopen('http://api.openstreetmap.org/api/0.6/map?bbox='
+                              + str(box)[1:-1].replace(" ", ""))
+
+    osm = open('map.osm','w')
```
Add space after the comma

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+                                    roadName = (typeHighway +
+                                                "_" +
+                                                str(self.data[i].get("data")
+                                                           .get("id")))
```
Remove one or two tabs

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+                                if roadName is None:
+                                    roadName = (typeHighway +
+                                                "_" +
+                                                str(self.data[i].get("data")
```
That line should be <= 79 chars

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+                                    if "node" in self.data[j].get("type"):
+
+                                        if (self.data[j].get("data")
+                                                   .get("id")) in node_ref:
```
Remove one tab

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+
+                            if "node" in self.data[j].get("type"):
+
+                                if self.data[j].get("data").get("id") in node_ref:
```
That line should be <= 79 chars

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+
+                                    coords = np.append(coords,
+                                                       self.data[j].get("data")
+                                                              .get("lon"))
```
Line over-indented. My pep8 checker complains here.

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+                                                              .get("lon"))
+                                    coords = np.append(coords,
+                                                       self.data[j].get("data")
+                                                              .get("lat"))
```
Line over-indented. My pep8 checker complains here.

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+
+                            if "node" in self.data[j].get("type"):
+
+                                if self.data[j].get("data").get("id") in node_ref:
```
Line too long.

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+
+                                    coords = np.append(coords,
+                                                       self.data[j].get("data")
+                                                              .get("lon"))
```
Line over-indented. My pep8 checker complains here.

caguero commented on osm2dict.py:
```diff
@@ -0,0 +1,308 @@
+                                                              .get("lon"))
+                                    coords = np.append(coords,
+                                                       self.data[j].get("data")
+                                                              .get("lat"))
```
Line over-indented. My pep8 checker complains here.

caguero commented:
I've added some minor pep8 comments.

caguero commented:
Related with the command line mode. I'm envisionning something like this:
gz_osm [options] <startLat> <startLong> <endLat> <endLong>

Options:
 -v [ --version ] Output version information.
 -h [--help]       Produce this help message.
 -a [--all]          Import all the objects.
 -b [--buildings] Import buildings.
 -d [--destination] Set the destination directory where the .sdf file will be generated.
 -m [--map]      Generate a .png file with the map.
 -r [--roads]      Import roads.
 -s [--sidewalks] Import sidewalks.
 -t [--traffic-lights] Import traffic lights.

Commit:
added test files

Tashwin committed:
Delete dict2sdf.pyc

Tashwin committed:
Delete example.py

Commit:
Deleted extra files

Commit:
Merge branch 'master' of https://github.com/Tashwin/gazebo_osm-1

Tashwin commented:
Thanks. I fixed the pep8 inconsistencies, added a few test files. 
I am currently working on the argparse functionality.

Commit:
added command line arguments

scpeters commented:
I'm having trouble running the gz_osm.py command with no arguments (which should maybe return a help string?):

```
$ ./gz_osm.py 
Traceback (most recent call last):
  File "./gz_osm.py", line 45, in <module>
    getMapImage(args.osmFile, args.imageFile)
TypeError: getMapImage() takes exactly 1 argument (2 given)
```

Commit:
added comand line support

Tashwin commented:
I forgot to update the getMapImage.py. It should work now.

Commit:
updated errors; updated road widths

Commit:
added features

Tashwin commented:
I  have added some new features to the maps.
The sdf file which is generated works well for smaller areas, but for larger areas, I am getting the following error from gazebo.

```
$ gazebo outFile.sdf 
Gazebo multi-robot simulator, version 2.2.1
Copyright (C) 2012-2013 Open Source Robotics Foundation.
Released under the Apache 2 License.
http://gazebosim.org

Msg Waiting for master
Msg Connected to gazebo master @ http://127.0.0.1:11345
Msg Publicized address: 192.168.1.62
Gazebo multi-robot simulator, version 2.2.1
Copyright (C) 2012-2013 Open Source Robotics Foundation.
Released under the Apache 2 License.
http://gazebosim.org

Msg Waiting for master
Msg Connected to gazebo master @ http://127.0.0.1:11345
Msg Publicized address: 192.168.1.62
Error [ConnectionManager.cc:565] Unable to get namespaces from master
Warning [Publisher.cc:134] Queue limit reached for topic /gazebo/default/model/info, deleting message. This warning is printed only once.
```

I have run gzsdf check <filename> which returns a 'Success' value.

Could you point me as to where I might be going wrong?

Commit:
updated README

Commit:
updated README

Commit:
added functionality to gz_osm.py

Commit:
updated files

Commit:
updated files

Commit:
updated sorting in osm2dict.py file

Commit:
added test for sdf files

Commit:
added comments

caguero commented:
:+1:  We should start thinking merging this pull request

Tashwin commented:
Yes, I added the comments to the files. Also, added support for a directly input osm file. I am working on the footway problem and if I can't solve it by tomorrow I'll create an issue regarding it.

scpeters commented:
Yes, let's merge and then use the issue tracker for future improvements.
